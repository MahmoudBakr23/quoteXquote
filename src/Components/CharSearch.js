/* eslint-disable react/prop-types */
import React from 'react';

const CharSearch = ({ handleSearch }) => (
  <div className="search-select">
    <input type="text" name="search" onChange={(e) => handleSearch(e)} placeholder="search..." />
    <ul className="hint">
      <li><h6>Naruto</h6></li>
      <li><h6>Hisoka</h6></li>
      <li><h6>Luffy</h6></li>
      <li><h6>Erin</h6></li>
      <li><h6>Meliodas</h6></li>
    </ul>
  </div>
);

export default CharSearch;
