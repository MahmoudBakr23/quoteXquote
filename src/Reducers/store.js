import { createStore } from 'redux';
import animeReducer from './anime';

const store = createStore(animeReducer);

export default store;
