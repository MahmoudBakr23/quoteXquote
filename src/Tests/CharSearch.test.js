import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Reducers/store';
import CharSearch from '../Components/CharSearch';

describe('Test Anime List Component', () => {
  const anime = render(<CharSearch />);

  test('Testing elements', () => {
    anime.getByText('Hisoka');
    anime.getByText('Meliodas');
    anime.getByText('Erin');
  });

  it('Expect to render Character Search component', () => {
    const result = renderer.create(
      <Provider store={store}>
        <CharSearch />
      </Provider>,
    ).toJSON;
    expect(result).toMatchSnapshot();
  });
});
