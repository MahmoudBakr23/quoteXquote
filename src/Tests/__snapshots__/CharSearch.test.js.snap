// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Anime List Component Expect to render Character Search component 1`] = `[Function]`;
